pygame/rwobject.py

```diff
@@ -17,7 +69,18 @@ def rwops_encode_file_path(filepath):
 
 
 def rwops_from_file(fileobj):
-    rwops = sdl.SDL_RWFromFP(fileobj, 0)
+    try:
+        # We try use the SDL helper first, since
+        # it's the simplest code path
+        rwops = sdl.SDL_RWFromFP(fileobj, 0)
+    except (TypeError, IOError):
```

Comment:
pygame doesn't ever use SDL_RWFromFP - actual file objects go through the same code path as everything else. Since we know that the SDL_RWFromFP code path works well, and adding multiple extra levels of indirection isn't the best idea ever, I've opted not to duplicate pygame's logic here and keep the SDL_RWFromFP option.

Catching multiple exceptions is needed because of differences between python versions:

StringIO.StringIO has no fileno method, so cffi raises a TypeError because it figure out how to covert it to a file.

io.BytesIO and io.StringIO, however, do have a fileno method. This mehod raises an IOError if the underlying buffer doesnt have an associated file (it actually raises io.UnsupportedOperation, which subclasses both ValueError & IOError, but the docs just say IOError, so I've chosen to follow the documented behaviour).